### PR TITLE
Updater topic name can be remapped via CMAKE

### PR DESCRIPTION
**Pull request by bjv-capra**
closes https://github.com/micro-ROS/micro_ros_diagnostics/issues/19

@norro I took the liberty of creating a PR with what we discussed.

- [x] Update documentation
- [x] move example_analyzers
- [x] rename to `output_topics`
- [x] rename to `input_topics`

### CODE REVIEW

**Review by norro — APPROVED** [review]
LGTM, thanks for the PR.

Should be complemented with https://github.com/micro-ROS/micro_ros_diagnostics/pull/21 soon to allow the bridge to work with modified topics.

**Comment by bjv-capra** [comment]
If it's okay with you, I could try to cover both (bridge and updater) on this PR.

**Comment by bjv-capra** [comment]
@norro This PR is ready for review (Still needs documentation updates)

What I did is to 

- remove the FQDN from the topics in the bridge

```
`/diagnostics` -> `diagnostics`
`/diagnostucs_uros` -> `diagnostics`
```
- added arguments to the launch file to be able to modify the namespace as well as to remap the topics.
- Made the hardcoded topic on the `updater` to be configurable via CMakeLists

Let me know what you think

File: micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py [hunk 1]
```diff
@@ -31,10 +31,27 @@ def generate_launch_description():
             'log_level',
             default_value=['info'],
             description='Logging level'),
+        launch.actions.DeclareLaunchArgument(
+            'remap_diagnostics_uros',
+            default_value=['diagnostics_uros'],
+            description='Namespace for diagnostics uros topic'),
```
Comment (bjv-capra) [inline micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py:37]: Need to reword this

File: micro_ros_diagnostic_updater/CMakeLists.txt [hunk 2]
```diff
@@ -29,6 +29,7 @@ option(MICRO_ROS_DIAGNOSTIC_UPDATER_EXAMPLES "Build example updaters for the mic
 
 # Configuration
 set(MICRO_ROS_DIAGNOSTIC_UPDATER_MAX_TASKS_PER_UPDATER "5" CACHE STRING "Number of tasks preallocated per updater.")
+set(MICRO_ROS_DIAGNOSTIC_UPDATER_DIAGNOSTICS_TOPIC_PREFIX "" CACHE STRING "Prefix for micro-ROS diagnostic topic (append slash '/')")
```
Comment (bjv-capra) [inline micro_ros_diagnostic_updater/CMakeLists.txt:32]: Need to remove this

File: micro_ros_diagnostic_updater/src/config.h.in [hunk 3]
```diff
@@ -2,5 +2,6 @@
 #define MICRO_ROS_DIAGNOSTIC_UPDATER_CONFIG_H
 
 #define MICRO_ROS_DIAGNOSTIC_UPDATER_MAX_TASKS_PER_UPDATER @MICRO_ROS_DIAGNOSTIC_UPDATER_MAX_TASKS_PER_UPDATER@
+static const char UROS_DIAGNOSTIC_UPDATER_TOPIC[] = "@MICRO_ROS_DIAGNOSTIC_UPDATER_DIAGNOSTICS_TOPIC_PREFIX@diagnostics_uros";
```
Comment (bjv-capra) [inline micro_ros_diagnostic_updater/src/config.h.in:5]: Need to remove this

File: micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py [hunk 4]
```diff
@@ -31,10 +31,27 @@ def generate_launch_description():
             'log_level',
             default_value=['info'],
             description='Logging level'),
+        launch.actions.DeclareLaunchArgument(
+            'remap_diagnostics_uros',
```
Comment (norro) [inline micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py:35]: Maybe ìnput_topics`? Already preparing for the fact that the bridge should listen to x topics at some point.

File: micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py [hunk 5]
```diff
@@ -31,10 +31,27 @@ def generate_launch_description():
...
+            default_value=['diagnostics_uros'],
+            description='Namespace for diagnostics uros topic'),
+        launch.actions.DeclareLaunchArgument(
+            'remap_diagnostics',
+            default_value=['diagnostics'],
```
Comment (norro) [inline micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py:40]: Maybe `output_topic`?

**Review by norro — CHANGES_REQUESTED** [review]
Looks good to me. Could you also move the example_analyzers file into the examples folder?

File: micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py (hunk 4 above)
Comment (bjv-capra) [reply micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py:35]: I'll make the changes

Comment (bjv-capra) [reply micro_ros_diagnostic_bridge/launch/diagnostic_bridge.launch.py:35]: After thinking about it, I think we should stick to input_topic, and later decide what do we do about it when we have multiple. As there are multiple options, like providing two separate arguments. Or simply deprecating the single one.

What do you think?

**Comment by bjv-capra** [comment]
@norro this PR is ready

**Comment by bjv-capra** [comment]
@norro friendly ping. I can't merge until you remove your "changes requested" :)

**Review by norro — APPROVED** [review]
Sorry, Bart, and thanks for the friendly ping. I tested it according to the documentation and it works as expected.
Thanks for the PR!

**Pull request merged**